march_hardware/include/march_hardware/joint.h

```diff
@@ -139,8 +89,8 @@ class Joint
   double absolute_position_ = 0.0;
   double velocity_ = 0.0;
 
-  std::unique_ptr<IMotionCube> imc_ = nullptr;
-  std::unique_ptr<TemperatureGES> temperature_ges_ = nullptr;
+  std::shared_ptr<MotorController> controller_ = nullptr;
+  std::shared_ptr<TemperatureGES> temperature_ges_ = nullptr;
```

Comment:
Because I now pass a list of ethercat slaves to ethercat_master (so that march_robot does not need to know which joints are ethercat slaves, and because it makes more sense anyway). So now there are two or three pointers to a a Ges, on in the EthercatMaster, and one in one or two of the joints (previously, the two joints had two separate ges objects for the same ges).